### PR TITLE
Update "I/O: FITS and ASCII" for AAS 235

### PR DESCRIPTION
**Pull request by pllim**
Here are my edits in preparation for AAS 235, which include but not limited to:

* Integrate `astropy.visualization` into plotting.
* Favor `QTable` API over low-level access using `astropy.io.fits` or `astropy.io.ascii`.
* Always use `with` context manager to open file.
* Avoid using `plt.<command>` directly. While it shows separate plots nicely in a notebook, it behaves differently in Python or IPython sessions. Using the `fig` and `ax` combo is more portable.
* Remove `! cat <filename>` cell as it will certainly break on someone's Windows machine.
* Remove the FITS exercise for 3 reasons:
  * No solution is provided.
  * It is asking user to do something that was never covered.
  * We won't have enough time anyway. And if we somehow do, it would be a good opportunity to play catch up in the workshop, since some other sections are going to go over.
* Remove unused data file.
* Added section links to table of contents.
* Miscellaneous code and text clean-ups.

### CODE REVIEW

File: 05-FITS/io_fits_ascii_overview.ipynb [hunk 1]
```diff
@@ -331,27 +338,17 @@
...
-    "plt.imshow(scidata, vmin=50, vmax=200)\n",
-    "plt.colorbar()"
-   ]
-  },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "For astronomical images, you often want the image scaling to be logarithmic. You can do this a number of ways, but the recommended was is to define the `norm` argument to be `matplotlib.colors.LogNorm()`."
+    "For astronomical images, you can use `astropy.visualization` to [normalize and stretch](https://docs.astropy.org/en/stable/visualization/normalization.html) the display. Here, we want it to be logarithmic."
```
Comment (stargaser) [inline 05-FITS/io_fits_ascii_overview.ipynb]: Interesting to introduce `astropy.visualization` here. It is shown in the WCS and Images section. I think there's no need to modify the notebook. I suggest to mention that more details will be shown in the later session.

File: 05-FITS/io_fits_ascii_overview.ipynb [hunk 2]
```diff
@@ -646,9 +644,12 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "plt.plot(throughput_data['lambda'], throughput_data['throughput'])\n",
-    "plt.xlabel('Wavelength ($\\mu$m)')\n",
-    "plt.ylabel('Throughput')"
+    "from astropy.visualization import quantity_support\n",
```
Comment (stargaser) [inline 05-FITS/io_fits_ascii_overview.ipynb]: Another good use of astropy.visualization (and a new feature I have haven't used before).

File: 05-FITS/io_fits_ascii_overview.ipynb [hunk 3]
```diff
@@ -6,20 +6,21 @@
...
-    "    - Image data\n",
-    "    - Table data\n",
+    "    - <a href=\"#Working-with-FITS-files\">Open a FITS file</a>\n",
+    "    - <a href=\"#Working-with-headers\">Examine its header</a>\n",
+    "    - <a href=\"#Working-with-image-data\">Image data</a>\n",
+    "    - <a href=\"#Working-with-FITS-tables\">Table data</a>\n",
     "    - Visualizing with `matplotlib`\n",
     "- Learn how to use ASCII files\n",
-    "    - Open an ASCII file\n",
+    "    - <a href=\"#Working-with-ASCII-files\">Open an ASCII file</a>\n",
```
Comment (stargaser) [inline 05-FITS/io_fits_ascii_overview.ipynb]: Maybe "Visualizing with matplotlib" should mention `astropy.visualization`?

File: 05-FITS/io_fits_ascii_overview.ipynb [hunk 4]
```diff
@@ -53,9 +54,16 @@
...
+    "with fits.open('j94f05bgq_flt.fits.gz') as f:\n",
+    "    f.info()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Opening a FITS file using the Python *with* statement\n",
+    "ensures that the file is closed without explicitely doing so."
```
Comment (stargaser) [inline 05-FITS/io_fits_ascii_overview.ipynb]: explicitly

Comment (pllim) [reply 05-FITS/io_fits_ascii_overview.ipynb]: Good catch!

**Comment by pllim** [comment]
@stargaser , I think I have addressed all your comments? If so, please merge it. Thanks!

File: 05-FITS/io_fits_ascii_overview.ipynb [hunk 5]
```diff
@@ -480,94 +475,98 @@
...
    "metadata": {},
-   "outputs": [],
    "source": [
-    "with open('throughput.txt') as f:\n",
-    "    throughput_data = f.read()\n",
-    "throughput_data"
+    "What if your data is in a text file? We've included `throughput.txt` as an example.\n",
+    "\n",
+    "    Detector throughput values\n",
+    "    â€”â€”\n",
```
Comment (stargaser) [inline 05-FITS/io_fits_ascii_overview.ipynb]:
```suggestion
    "    --\n",
```

File: 05-FITS/io_fits_ascii_overview.ipynb [hunk 6]
```diff
@@ -165,21 +168,22 @@
...
     "# Assigning a tuple to a keyword splits\n",
     "# the tuple into a keyword value and comment.\n",
+    "# FITS header keyword is also not case-sensitive.\n",
     "    \n",
-    "f[1].header['NOBS'] = (2, \"Number of observing nights\")\n",
-    "print(\"observer:\", f[1].header['observer'])\n",
-    "print(f[1].header.comments['NOBS'], f[1].header['NOBS'])"
+    "f[extnum].header['NOBS'] = (2, \"Number of observing nights\")\n",
+    "print(\"observer:\", f[extnum].header['OBSERVER'])\n",
+    "print(f[extnum].header.comments['NOBS'], f[extnum].header['NOBS'])"
```
Comment (stargaser) [inline 05-FITS/io_fits_ascii_overview.ipynb]: I feel we ought to mention `header.insert` when one cares about the ordering of keywords.

File: 05-FITS/io_fits_ascii_overview.ipynb [hunk 7]
```diff
@@ -207,12 +211,15 @@
...
    "source": [
+    "import time\n",
+    "\n",
     "with fits.open('j94f05bgq_flt.fits.gz', mode='update') as f:\n",
-    "    f[1].header[\"history\"] = \"New history card.\"\n",
-    "    f[1].header[\"comment\"] = \"This is a 47 Tuc observation with HST\"\n",
-    "    f[1].header[\"comment\"] = \"I'd like to observe it with JWST.\"\n",
-    "    f[1].header['comment']\n",
-    "    print(f[1].header['comment'])"
+    "    f[extnum].header[\"history\"] = \"{} New history card.\".format(time.ctime())\n",
```
Comment (stargaser) [inline 05-FITS/io_fits_ascii_overview.ipynb]: I feel that we should tell people about `header.add_comment()` and `header.add_history()` and that these should be presented as best practices.

File: 05-FITS/io_fits_ascii_overview.ipynb [hunk 8]
```diff
@@ -136,25 +143,21 @@
...
-    "    print(\"Updated crpix1 :\", f[1].header['crpix1'])"
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "metadata": {},
-   "source": [
-    "Opening a FITS file using the Python *with* statement\n",
-    "ensures that the file is closed without explicitely doing so."
+    "    value = f[extnum].header[keyname]\n",
```
Comment (stargaser) [inline 05-FITS/io_fits_ascii_overview.ipynb]: `header.get()` does not raise the KeyError IIRC and it also allows the specification of a default (which defaults to `None` if the keyword is not in the header). I would like us to show that as best practice.

File: 05-FITS/io_fits_ascii_overview.ipynb (hunk 5 above)
Comment (pllim) [reply 05-FITS/io_fits_ascii_overview.ipynb]: I copy-paste this from the data file. Are you seeing something different when you open `throughput.txt`? I wonder if the encoding is OS-specific.

Comment (pllim) [reply 05-FITS/io_fits_ascii_overview.ipynb]: OK, nvm... Now I see the dashes. :woman_shrugging:

**Comment by pllim** [comment]
@stargaser , I think I got all the commends now?

**Pull request merged**